src/main/java/filemanager/directorytracker/ScheduleFileDirectory.java

```diff
@@ -0,0 +1,43 @@
+package filemanager.directorytracker;
+
+
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Injector;
+import filemanager.binder.FileServiceInjector;
+import filemanager.service.ConverterFromJsonToXmlService;
+import filemanager.serviceImpl.ConverterFromJsonToXmlServiceImpl;
+
+import java.io.IOException;
+import java.nio.file.*;
+
+public class ScheduleFileDirectory extends TrackerFileDirectory {
+
+    @Inject
+    private ConverterFromJsonToXmlService converter;
+
+    public ScheduleFileDirectory() {
+        init();
+    }
+
+    @Override
+    public void goThroughToCheckFile() {
+        Injector injector = Guice.createInjector(new FileServiceInjector());
+        converter = injector.getInstance(ConverterFromJsonToXmlServiceImpl.class);
+        try {
+            Files.walk(Paths.get(rootFolder + environment), FileVisitOption.FOLLOW_LINKS).filter(Files::isRegularFile).forEach(path -> {
```

Comment:
This class should take execution time and iterate through files that are present in related directory. This logic could be covered with tests.